### PR TITLE
Hari

### LINKED ISSUE #22: passengers_drop
Dropping the passengers at requested floors

### LINKED ISSUE #23: create_floor_array
Creates an array of floors of size entered by the user

### PR DESCRIPTION
**Pull request by Hariprasadm1408**
Moving 2 function and a test function as part of release 1

### CODE REVIEW

File: src/passengers_drop.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ **/
+int passengers_drop(struct elevator elevator_arr[], int index, struct passenger* floor_arr[], int cur_floor)
+{
+
+	assert(index <= elevator_count);
+	assert(cur_floor <= floor_count);
+	assert(cur_floor != elevator_arr[index].cur_floor);
+
+	
+	int passenger_dropped = elevator_arr[index].passenger_count;
```
Comment (MaazJamal) [inline src/passengers_drop.c]: This should be max passengers. As we go over all the passengers

File: src/create_floor_array.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * The function create arrays for every floor
+ *
+ * Gets the number of floors as input and creates a list of array for every floor.
+ *
+ * @return structure of arrays for the given number of floors.
+ *  */
+ struct passenger * create_floor_array(int no_of_floors)
+ {
+    if((no_of_floors==0)||(no_of_floors>100)||(no_of_floors<1))
```
Comment (MaazJamal) [inline src/create_floor_array.c]: The number of floors isn't fixed. it should be no_of_floors>floor_count.

File: src/passengers_drop.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    //Loop to iterate the overall passengers in the elevator
+	for(int i = 0; i < passenger_capacity; i++)
+	{
+		if(elevator_arr[index].passenger_arr[i].in_elevator)
+		{
+
+			if(floor_arr[cur_floor] != NULL && elevator_arr[index].passenger_count < passenger_capacity)
```
Comment (MaazJamal) [inline src/passengers_drop.c]: You need to check if elevator_arr[index].passenger_arr[i].dest_floor == cur_floor. i.e this is the floor the passenger need to get off at. and then simply set 
    elevator_arr[index].passenger_arr[i].in_elevator = false;
    elevator_arr[index].passenger_count--;

File: src/passengers_drop.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+			{
+			elevator_arr[index].passenger_arr[i] = *(floor_arr[cur_floor]);
+			elevator_arr[index].passenger_arr[i].in_elevator = false; //Checks the status
+			elevator_arr[index].passenger_count--;
+			passenger_dropped++;
+			remove_passenger_floor(floor_arr,cur_floor); //removes the apsser from floor array
```
Comment (MaazJamal) [inline src/passengers_drop.c]: The passenger was already removed from floor when passenger was added to the elevator.

File: src/passengers_drop.c (hunk 1 above)
Comment (Hariprasadm1408) [reply src/passengers_drop.c]: Changes applied.

**Review by MaazJamal — APPROVED** [review]
OKay i think this should be fine now.

File: src/count_passenger.c [hunk 5]
```diff
@@ -17,6 +17,11 @@ int count_passenger()
     FILE *f;
     char a[10];
     f=fopen("datafolder/input.txt","r");
+    if (f==NULL)
```
Comment (divyalakshmi-padmanaban) [inline src/count_passenger.c]: f==NULL condition captures any errors in opening input files.
There can be a possibility of empty input file in specified path .
I think,there has to be a check for this too.

Comment (Hariprasadm1408) [reply src/count_passenger.c]: Empty file will return count as zero. Its handled.

**Pull request merged**